src/main/java/com/google/firebase/FirebaseApp.java

```diff
@@ -597,7 +597,18 @@ private static FirebaseOptions getOptionsFromEnvironment() throws IOException {
       parser = jsonFactory.createJsonParser(reader);
     }
     parser.parseAndClose(builder);
-    builder.setCredentials(GoogleCredentials.getApplicationDefault());
+    builder.setCredentials(getApplicationDefaultOrEmptyCredentials());
     return builder.build();
   }
+
+  private static GoogleCredentials getApplicationDefaultOrEmptyCredentials() {
+    try {
+      return GoogleCredentials.getApplicationDefault();
+    } catch (IOException e) {
+      logger.error(
+          "Failed to fetch default application credentials. Proceeding with blank credentials.",
```

Comment:
Nit: Prefer "empty" over "blank". Your call.

Reply:
Changed this line as we return null now, wrapped in `DeferredApplicationCredentials`. Feel free to suggest another change if required, but I feel the NPE will be self explanatory.